### ISSUE TITLE
Undo removes the partial output of currently running cell; cannot redo changes undone when other cell produces outputs

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#10694]
## Description

The new system of storing history of changes in 3.1 introduced with Real Time Collaboration (RTC) has some peculiarities which do not work great with my [workflow](https://xkcd.com/1172/). The synchronization of outputs and inputs might be neat in some cases, but it can be disruptive for live-editing code while the cell is already running.

### Demo isolated to a single cell

![undo-affects-output](https://user-images.githubusercontent.com/5832902/126876206-6b825cc4-ffda-4486-9eef-07fd4c21530c.gif)

1. Open notebook using JupyterLab 3.1rc2
2. Execute following code:
    ```python
    from time import sleep
    
    for i in range(20):
        print(i)
        sleep(1)
    ```
3. Try pressing <kbd>Ctrl</kbd> + <kbd>Z</kbd> while the code is running
4. See that the last bit of output gets removed
5. Try to make a change in the code and undo it with <kbd>Ctrl</kbd> + <kbd>Z</kbd>
6. See that it will be only undone after removing the bits of output that showed up since making the change in the code

### The real problems

1) I cannot work in other cell when I use a progress reporting in an earlier cell or I will loose bits of output:

![Working in other cell disrupted](https://user-images.githubusercontent.com/5832902/126876524-1acd60de-e35f-42c6-bb07-3ce371cdccec.gif)

2) I cannot redo changes that I undo:

![cannot-redo](https://user-images.githubusercontent.com/5832902/126876533-8cc83ba1-202d-40aa-ac02-f86fa2aa887d.gif)

## Expected behavior

I was able to get used to the undo being document-wide rather than cells-specific, but it still can be annoying from time to time.

I think that there are two issues:
a) allowing users to disable the document-wide history tracking when not in RTC mode
b) separating history of outputs from history of inputs, at least when not in RTC mode

## Context

- Browser and version: Chrome
- JupyterLab version: 3.1.0rc2

**Comment by goanpeca** [comment]
Thoughts on this one @dmonad  ?

**Comment by hbcarlos** [comment]
Actually, there is a way of preventing Yjs from capturing operations.
see: https://docs.yjs.dev/api/undo-manager#example-stop-capturing

I'm working on this. I'll open a PR.

**Comment by mlucool** [comment]
> a) allowing users to disable the document-wide history tracking when not in RTC mode

This would be great to do. To some users, this behavior is detrimental to their workflow.

**Comment by vidartf** [comment]
> b) separating history of outputs from history of inputs, at least when not in RTC mode

I don't think having undo/redo of outputs makes sense. While it might make sense to keep the history, I think it should be marked in some way to not be included in the undo/redo stream.

**Comment by krassowski** [comment]
Closing as fixed in #10756 (released in 3.1.2). I opened #10791 to track allowing to disable the history tracking entirely.
